ukfom/ukf.hpp

```diff
@@ -187,32 +187,21 @@ class ukf {
 	{
 		assert(X.size() == 2 * n + 1);
 
-		lapack::cholesky<n> L(sigma);
+        Eigen::LLT<cov> lltOfSigma(sigma); // compute the Cholesky decomposition of A
+        cov L = lltOfSigma.matrixL(); // retrieve factor L  in the decomposition
 
-		if (!L.isSPD())
-		{
-			std::cerr << std::endl << "sigma is not SPD:" << std::endl
-					  << sigma << std::endl
-					  << "---" << std::endl;
-			Eigen::EigenSolver<cov> eig(sigma);
-			std::cerr << "eigen values: " << eig.eigenvalues().transpose() << std::endl;
-		}
-		
-		assert(L.isSPD());
```

Comment:
Semi-positive definite is not sufficient for an `Eigen::LLT` decomposition. It must be really positive definite.
On could fall-back to making a full (selfadjoint) Eigenvalue decomposition and use `L=SelfAdointEigenSolver::operatorSqrt()` as alternative (this requires setting negative eigenvalues to 0.0).